src/game.rs

```diff
@@ -131,6 +138,16 @@ impl GameState {
                 }, UpdateEndState::Lost => {
                     break;
                 }
+                UpdateEndState::Pause => {
+                    // Keep the game paused until 'p' is pressed again
+                    loop {
```

Comment:
Can you make this a while loop?

```
while inp.get_key() != b'p' {
    ...
}
```

Also, the actions in the pause sub-loop are pretty self-explanatory, so there's no need for any comments